### PR TITLE
Update the advanced features documentation

### PR DESCRIPTION
**Pull request by RobinPicard**
In the "Features" section of our documentation, we want to include an "Advanced" subsection that would contain a page on logits processors. This page quickly goes over what logits processors are, why one would want to create a custom one and gives a short but functional example.

### CODE REVIEW

File: docs_new/features/advanced/logits_processors.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+Processors can be used to:
+1. Generate structured output (e.g., JSON that follows a specific schema)
+2. Prevent the model from generating specific words or tokens
+3. Implement custom token sampling strategies
+
+## Overview
+
+Outlines uses logits processors with steerable models—models that run locally and allow fine-grained control over the generation process. When using such models in Outlines, the output type provided is turned into a logits processor that is then passed to the inference engine.
+
+There are four models for which we rely on logits processors:
```
Comment (cpfiffer) [inline docs_new/features/advanced/logits_processors.md:17]:
```suggestion
There are four models that support logits processors:
```

File: docs_new/features/advanced/logits_processors.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+```python
+from outlines import Generator
+
+model = ...
+logits_processor = ...
+
+# Create a generator with a custom logits processor
+generator = Generator(model, processor=logits_processor)
+response = generator("Hello world")
```
Comment (cpfiffer) [inline docs_new/features/advanced/logits_processors.md:36]: Could we add a specific example here?

**Review by cpfiffer — APPROVED** [review]
